Makefile

```diff
@@ -1,37 +1,41 @@
-# mark targets to avoid confusion with files
-.PHONY: clean dir help java node
-all: clean java node
-
-JAVA_DIR = ./java/src/main/gen
-PROTO_DIR = ./api
+DIR = $(shell pwd)
+JAVA_DIR = ./java/src/main/java
 NODE_DIR = ./node
-PROTOC_BIN ?= protoc
-ALL_PROTO_FILES = $(shell find api -iname "*.proto" -exec echo -n '"{}" ' \; | tr '\n' ' ')
 
-dir: ## Creates outdir if necessary
-	[ -d $(JAVA_DIR) ] || mkdir -p $(JAVA_DIR)
+OUTPUT ?= $(JAVA_DIR)
+LANGUAGE ?= java
+GRPCPLUGIN ?= /usr/bin/protoc-gen-grpc-$(LANGUAGE)
+# with https://github.com/TheThingsIndustries/docker-protobuf/issues/9 we could use this image also for TypeScript
+PROTOC ?= docker run --rm -v $(DIR):$(DIR) -w $(DIR) thethingsindustries/protoc
 
-node:  ## Generates node resources
-	@echo "+ $@"
-	$(PROTOC_BIN) --proto_path=$(PROTO_DIR) \
-		--plugin=protoc-gen-ts=$(NODE_DIR)/node_modules/.bin/protoc-gen-ts \
-		--plugin=protoc-gen-grpc=$(NODE_DIR)/node_modules/.bin/grpc_tools_node_protoc_plugin \
-		--js_out=import_style=commonjs,binary:$(NODE_DIR) \
-		--ts_out=service=grpc-node:$(NODE_DIR) \
-		--grpc_out=$(NODE_DIR) $(ALL_PROTO_FILES)
+FLAGS+= --proto_path=.
+ifeq ($(LANGUAGE),node)
+FLAGS+= --plugin=protoc-gen-ts=$(NODE_DIR)/node_modules/.bin/protoc-gen-ts
+FLAGS+= --plugin=protoc-gen-grpc=$(NODE_DIR)/node_modules/.bin/grpc_tools_node_protoc_plugin
+FLAGS+= --js_out=import_style=commonjs,binary:$(NODE_DIR)
+FLAGS+= --ts_out=service=grpc-node:$(NODE_DIR)
+else
```

Comment:
Hier vielleicht ein ifeq Java statt dem else ? Dann ist es leichter erweiterbar und Java nicht schon wieder der default.

Reply:
currently only `TypeScript` is the exception (see linked issue) - other languages (java, ruby, js, php, swift) are fine with the `else` branch